ui/app/components/database-role-edit.js

```diff
@@ -115,6 +121,20 @@ export default class DatabaseRoleEdit extends Component {
       const { mode, model } = this.args;
       if (!this.isValid()) return;
 
+      // if we're creating and rotating a static role immediately, warn user
+      if (!model.skip_import_rotation && model.type === 'static' && mode === 'create') {
+        this.saveIssuerWarning =
+          "You have enabled 'Rotate immediately' for this static role. \n Vault will update the password immediately after you save. \n \n NOTE: This will disrupt any active use of this role outside of Vault.";
+        return;
+      }
+      await this.saveRole.perform();
+    })
+  );
+
+  saveRole = task(
```

Comment:
now that the save task has been separated, `handleCreateEditRole` should probably be converted to just an async action unless we need any concurrent behaviors from the method.